### ISSUE TITLE
CDS type "Date" to TypeScript type "Date" conversion is misleading/incorrect

### ISSUE THREAD

**Issue by stockbal** [cap-js/cds-typer#18]
Hi,

first up, great to finally see some native type generation support in CAP. The automatic regeneration via VS Code extension is also working great 🥳 .

Unfortunately I stumbled upon an issue that I also remarked in the `cds2types` project (https://github.com/mrbandler/cds2types/issues/40). It concerns the handling/conversion of CDS `Date` types.  
I created a sample [repository](https://github.com/stockbal/cap-samples/tree/main/cap-type-generator) where the issue can be tested.

Basically the typed result of e.g. a `SELECT` query wrongly marks a CDS `Date` column as a JavaScript `Date` type wherein it is actually just a formatted date string.

Regards,
Ludwig

**Comment by daogrady** [comment]
Hi Ludwig,

just giving a quick update: I was able to get a hold of the runtime team to confirm what you have found. CDS `Date`, _can_ be a JS `Date` object under certain circumstances, but will be a `string` in most cases. Will change accordingly in an upcoming release.

Best,
Daniel

**Comment by stockbal** [comment]
Hi Daniel,

That's really funny. I was just about to suggest that we could maybe use a union type `string | Date`. 😉
Looking forward to it 👌.

Thanks and regards,
Ludwig

**Comment by stockbal** [comment]
Also fine. I just checked, and it looks good.
Thanks for the quick fix
